source/elements/oneDAL/source/algorithms/clustering/kmeans.rst

```diff
@@ -1,153 +1,494 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
 =======
 K-Means
 =======
 
-K-Means is a clustering algorithm that partitions :math:`n` observations into
-:math:`k` clusters minimizing some criterion within each cluster. Each cluster
-is characterized by a representative point, called *a centroid*, and a cluster
-radius.
+The K-Means algorithm partitions :math:`n` samples into :math:`k` *clusters*
```

Comment:
term 'samples' is used, and in terminology section we seem to introduce different